src/query/storage/m3/storage.go

```diff
@@ -110,9 +110,19 @@ func (s *m3storage) FetchRaw(
 		namespaces = s.clusters.ClusterNamespaces()
 		now        = time.Now()
 		fetches    = 0
-		result     multiFetchResult
 		wg         sync.WaitGroup
 	)
+	if len(namespaces) == 0 {
+		return nil, noop, fmt.Errorf("no namespaces configured")
+	}
+
+	pools, err := namespaces[0].Session().IteratorPools()
```

Comment:
maybe leave a todo to find a way where we're not always ignoring pools from the other namespaces

Reply:
talked offline, will comeback for this.